<launch>
    <arg name="way_file" default="optionTask1.csv"/>
    <arg name="sim" default="true"/>

    <!--<node name="odom_static" pkg="tf2_ros" type="static_transform_publisher"
      args="0 0 0 0 0 0 base_link odom"/>-->
    <node name="map_static" pkg="tf2_ros" type="static_transform_publisher"
      args="0 0 0 0 0 0 odom map"/>

    <group if="$(arg sim)">
        <include file="$(find beego_gazebo)/launch/empty_world.launch" >
            <arg name="rviz" value="false"/>
            <arg name="world_name" default="$(find yamabiko)/worlds/poll_pass.world"/>
        </include>
    </group>
    <group unless="$(arg sim)">
        <node name="ypspur_ros_bridge" pkg="ypspur_ros_bridge" type="ypspur_ros_bridge">
            <remap from="cmd_vel" to="diff_drive_controller/cmd_vel"/>
        </node>
        <node name="hokuyo" pkg="hokuyo_node" type="hokuyo_node" respawn="false" output="screen">
            <!-- Starts up faster, but timestamps will be inaccurate. --> 
            <param name="calibrate_time" type="bool" value="false"/> 
            <!-- Set the port to connect to here -->
            <param name="port" type="string" value="/dev/ttyACM0"/> 
            <param name="intensity" type="bool" value="false"/>
            <param name="frameid" value="hokuyo_link"/>
        </node>
        
    </group>

    <!-- obstacle detector -->
    <rosparam command="load" file="$(find yamabiko)/config/obstacle_detector.yaml" />
    <node name="obstacle_extractor" pkg="obstacle_detector" type="obstacle_extractor_node"/>
    <node name="obstacle_tracker" pkg="obstacle_detector" type="obstacle_tracker_node" />

    <node name="person_tracking" pkg="yamabiko" type="person_tracking" >
        <param name = "track_radius" value="0.5"/>
    </node>

    <node name="move_to_pose" pkg="yamabiko" type="move_to_pose" output="screen">
        <rosparam command="load" file="$(find yamabiko)/config/move_to_pose.yaml" />
        <remap from="move_to_pose/pose_in" to="person_tracking/pose_out"/>
    </node>

    <node name="twist_maneger" pkg="yamabiko" type="twist_maneger">
        <param name="base_link_frame_id" value="base_link"/>
        <param name="map_frame_id" value="map"/>
        <param name="loop_rate" value="30"/>
        <remap from="cmd_vel" to="move_to_pose/cmd_vel"/>
        <remap from="twist_maneger/cmd_vel" to="/diff_drive_controller/cmd_vel"/>
    </node>

    <!--rviz-->
    <include file="$(find yamabiko)/launch/include/model_show.launch" >
        <arg name="include" value="true"/>
    </include>

    <arg name="rvizconfig" default="$(find yamabiko)/rviz/midTask2.rviz" />
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

</launch>